api/python/t4/util.py

```diff
@@ -244,3 +245,50 @@ def get_local_registry():
 
 def get_remote_registry():
     return load_config().get('default_remote_registry', None)
+
+
+def quiltignore_filter(paths, ignore_rules, url_scheme):
+    """Given a list paths, filter out the paths which are captured by the given
+    ignore rules.
+
+    Args:
+        paths (list): a list or iterable of paths
+        ignore_rules (list): a list or iterable of ignore rules, in Unix shell
+            style wildcard format
+        url_scheme (str): the URL scheme, only the "file" scheme is currently
+            supported
+    """
+    if url_scheme == 'file':
+
+        files, dirs = set(), []
+        for path in paths:
+            if path.is_file():
+                files.add(path)
+            else:
+                dirs.append(path)
+
+        for ignore_rule in ignore_rules:
+            ignore_rule = os.getcwd() + '/' + ignore_rule
+
+            if dirs:
+                dir_idx = 0
+                while True:
+                    dir = dirs[dir_idx]
```

Comment:
dir is a built in function in python, might be good to avoid using as a var name.

Reply:
Done (and, OK, I'll hold to this convention in the future).